projects/angular-cesium/src/lib/angular-cesium-widgets/services/entity-editors/rectangles-editor/rectangles-editor.service.ts

```diff
@@ -0,0 +1,439 @@
+import { publish, tap } from 'rxjs/operators';
+import { Injectable } from '@angular/core';
+import { CesiumService } from '../../../../angular-cesium/services/cesium/cesium.service';
+import { MapEventsManagerService } from '../../../../angular-cesium/services/map-events-mananger/map-events-manager';
+import { BehaviorSubject, Observable, Subject } from 'rxjs';
+import { CesiumEvent } from '../../../../angular-cesium/services/map-events-mananger/consts/cesium-event.enum';
+import { PickOptions } from '../../../../angular-cesium/services/map-events-mananger/consts/pickOptions.enum';
+import { RectangleEditUpdate } from '../../../models/rectangle-edit-update';
+import { EditModes } from '../../../models/edit-mode.enum';
+import { EditActions } from '../../../models/edit-actions.enum';
+import { DisposableObservable } from '../../../../angular-cesium/services/map-events-mananger/disposable-observable';
+import { CoordinateConverter } from '../../../../angular-cesium/services/coordinate-converter/coordinate-converter.service';
+import { EditPoint } from '../../../models/edit-point';
+import { CameraService } from '../../../../angular-cesium/services/camera/camera.service';
+import { Cartesian3 } from '../../../../angular-cesium/models/cartesian3';
+import { RectanglesManagerService } from './rectangles-manager.service';
+import { RectangleEditorObservable } from '../../../models/rectangle-editor-observable';
+import { EditableRectangle } from '../../../models/editable-rectangle';
+import { RectangleEditOptions } from '../../../models/rectangle-edit-options';
+import { PointProps } from '../../../models/polyline-edit-options';
+import { LabelProps } from '../../../models/label-props';
+import { generateKey } from '../../utils';
+
+export const DEFAULT_RECTANGLE_OPTIONS: RectangleEditOptions = {
+  addPointEvent: CesiumEvent.LEFT_CLICK,
+  dragPointEvent: CesiumEvent.LEFT_CLICK_DRAG,
+  dragShapeEvent: CesiumEvent.LEFT_CLICK_DRAG,
+  allowDrag: true,
+  pointProps: {
+    color: Cesium.Color.WHITE.withAlpha(0.8),
+    outlineColor: Cesium.Color.BLACK.withAlpha(0.2),
+    outlineWidth: 1,
+    pixelSize: 13,
+    virtualPointPixelSize: 8,
+    show: true,
+    showVirtual: true,
+    disableDepthTestDistance: Number.POSITIVE_INFINITY,
+  },
+  rectangleProps: {
+    material: Cesium.Color.CORNFLOWERBLUE.withAlpha(0.4),
```

Comment:
Played with it and it looks good!
can you please support for rectangle outline props, and to add white outline as a default option.

Reply:
Will do

Reply:
@eitanfr Do you want me to add in an icon for the rectangle drawer in the demo app or will you guys take care of that?

Reply:
yes, i will be happy if could do that. 
somthing like https://www.iconfinder.com/icons/755913/design_draw_graphic_rectangle_shape_tool_icon

Reply:
Ok - I'll leave the icon up to you then!